server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -396,24 +398,46 @@ private void deleteSynonymsSetObjects(String synonymSetId, ActionListener<BulkBy
     }
 
     public void deleteSynonymsSet(String synonymSetId, ActionListener<SynonymsReloadResult<AcknowledgedResponse>> listener) {
-        deleteSynonymsSetObjects(synonymSetId, listener.delegateFailure((deleteObjectsListener, bulkByScrollResponse) -> {
-            if (bulkByScrollResponse.getDeleted() == 0) {
-                // If nothing was deleted, synonym set did not exist
-                deleteObjectsListener.onFailure(new ResourceNotFoundException("Synonym set [" + synonymSetId + "] not found"));
-                return;
-            }
-            final List<BulkItemResponse.Failure> bulkFailures = bulkByScrollResponse.getBulkFailures();
-            if (bulkFailures.isEmpty() == false) {
-                deleteObjectsListener.onFailure(
-                    new ElasticsearchException(
-                        "Error deleting synonym set: "
-                            + bulkFailures.stream().map(BulkItemResponse.Failure::getMessage).collect(Collectors.joining("\n"))
+
+        // Checks reloading the analyzer to understand usage. We can iterate on this solution to target the specific analyzer
+        reloadAnalyzers(synonymSetId, listener.delegateFailure((reloadListener, reloadResult) -> {
+            Map<String, ReloadAnalyzersResponse.ReloadDetails> reloadDetails = reloadResult.reloadAnalyzersResponse.getReloadDetails();
+            if (reloadDetails.isEmpty() == false) {
+                Set<String> indices = reloadDetails.entrySet()
+                    .stream()
+                    .map(entry -> entry.getValue().getIndexName())
+                    .collect(Collectors.toSet());
+                reloadListener.onFailure(
+                    new IllegalArgumentException(
+                        "Synonym set ["
+                            + synonymSetId
+                            + "] cannot be deleted as it is used in the following indices: "
```

Comment:
Maybe make this error message more of a CTA for how to resolve? 

```suggestion
                            + "] cannot be deleted until it is removed from the following indices: "
```

Reply:
Makes sense, adding that!